### PR TITLE
add dot_to_package fn like rprojroot

### PR DESCRIPTION
**Pull request by mpadge**
This enables `write_codemetar` to be called from anywhere within a package directory structure. It is essentially a hard-coded version of what `rprojroot` does to locate the root of an R package. Pretty sure it has no effects on anything else, as it's only evoked when `pkg = "."`.

``` r
library(codemetar) # current version prior to this PR
setwd (<path>/<to>/<local>/<codemetar>)
setwd("./R")

write_codemeta()
#> Error: File `./DESCRIPTION' does not exist
```
That's annoying because it does exist! With this PR:
``` r
devtools::load_all (".", export_all = FALSE) # Version with this PR
#> Loading codemetar
write_codemeta()
#> … Getting CRAN metadata from RStudio CRAN mirror
#> ✔ Got CRAN metadata!
#> codemetar has the highest opinion of this R package :-)
#> … Getting Bioconductor metadata
#> ✔ Got Bioconductor metadata!
#> … Getting sysreqs URL from sysreqs API
#> ✔ Got sysreqs URL from sysreqs API!
#> … Asking README URL from GitHub API
#> ✔ Got README URL!
#> … Asking README URL from GitHub API
#> ✔ Got README URL!
#> … Getting repo topics from GitHub API
#> ✔ Got repo topics!
```

<sup>Created on 2021-02-16 by the [reprex package](https://reprex.tidyverse.org) (v1.0.0)</sup>

That's better!

### CODE REVIEW

**Comment by mpadge** [comment]
@cboettig You're like going to have to opine here somewhere, because this PR attempts to make functionality less strictly dependent on exact `pkg` location, but you comment in #303 about overloading the `pkg` argument. I'm not sure what you mean there, since the `is_package` function is just
``` r
  all(c("DESCRIPTION", "NAMESPACE", "man", "R") %in% dir(path))  # where `path` is passed as `pkg`
```

That's clearly only looking for source directories, and is not overloaded. So to the extent that it is not, I would suggest you might be best off first merging this PR, then getting @ThierryO to extend PR #303 to ensure that the `pkg` argument is "taken into account" for all functions. If, on the other hand, `pkg` is actually overloaded elsewhere, then you'll need to do some deeper thinking about reconciling these two issues.

File: R/utils.R [hunk 1]
```diff
@@ -138,6 +138,36 @@ whether_provider_badge <- function(badges, provider_name) {
   )
 }
 
+# dot_to_package: convert pkg = "." to proper package path ------------------
+dot_to_package <- function(pkg = ".") {
+
+    # https://github.com/r-lib/rprojroot/blob/master/R/root.R#L115:
```
Comment (maelle) [inline R/utils.R:144]: Could you please add rprojroot as a dependency? We'd still be under 20 Imports.

File: R/write_codemeta.R [hunk 2]
```diff
@@ -58,6 +58,8 @@ write_codemeta <- function(
   codemeta_json <- "codemeta.json"
 
   # Things that only happen inside a package folder...
+  if (pkg == ".")
```
Comment (maelle) [inline R/write_codemeta.R:61]: Can you please add curly braces? I _think_ that's the style we use elswhere.

File: tests/testthat/test-utils.R [hunk 3]
```diff
@@ -29,3 +29,11 @@ testthat::test_that("example_file works", {
 
 })
 
+testthat::test_that("dot_to_package", {
+  p0 <- get_root_path("jsonld")
```
Comment (maelle) [inline tests/testthat/test-utils.R:33]: Could you use a package from Imports as example? I wonder whether using a package from Suggests is something we might regret later.

File: tests/testthat/test-utils.R [hunk 4]
```diff
@@ -29,3 +29,11 @@ testthat::test_that("example_file works", {
...
+  p <- file.path(p0, "R")
+  wd <- setwd(p)
```
Comment (maelle) [inline tests/testthat/test-utils.R:35]: Would you mind adding a Suggests dependency on withr and using `withr::local_dir()`?

**Review by maelle — CHANGES_REQUESTED** [review]
Thanks a ton @mpadge! This sounds like an useful feature.

**Pull request merged**